### ISSUE TITLE
Restoring from Keys should be more explicit.

### ISSUE THREAD

**Issue by juanpc2018** [m2049r/xmrwallet#872]
for example:
cake-tech/cakewallet monero.com
has:
View & Spend Keys, Public & Private = 4.
but m2049r/xmrwallet monerujo only has 2 Private + public address,

xmrwallet restore from keys should clarify:
View Key Private:

Spend Key Private:

for people moving from other wallet to xmrwallet, to avoid human error or wasting time.

**Comment by chaserene** [comment]
I agree. in places where a view or spend key is referred to, it should be specified whether it's a public or a private one. this includes places such as:

* opening screen -> "+" -> Restore view-only wallet
* other wallet creation/restoration flows
* wallet screen -> "⋮" -> Show secrets!

**Comment by m2049r** [comment]
@anhdres ideas?

**Comment by anhdres** [comment]
Are we talking about adding the word *Private* to every instance of *View Key* for consistency and teaching the user that they should not share it as if it were an address?

As in here for example?
![photo_2024-02-05_19-18-39](https://github.com/m2049r/xmrwallet/assets/31753665/552120ba-7f12-4f75-b2f9-46f7edb681b9)

If that the case, I don't see why not. It would read:
**Public Address**
**Private View Key**

We should probably even say *Public 4.. Address* or *Public Base Address* or whatever it's properly called. Or not, since technically every other address is called a *subaddress*. God knows I think we should all stop using 4.. addresses in every wallet, and then just call subaddresses **addresses** because it fucks with every new user's head.

**Comment by chaserene** [comment]
@anhdres Monero (in its current CryptoNote paradigm) has both public/private view keys and public/private *spend* keys. (good explainer by dEBRUYNE [here](https://monero.stackexchange.com/a/982).)

places in the wallet where I encountered them:

* opening screen -> "+" ->
  * Restore view-only wallet: private view key
  * Restore wallet from private keys: private view key, private spend key
* wallet screen -> "⋮" -> Show secrets! -> Detailed information: private view key, private spend key

in a popup/tooltip I also encountered the term "secret" instead of "private" for a key, which isn't wrong, but for clarity I would stick to public/private. this is also the terminology in Zero To Monero.

I would avoid calling an address public. addresses are by definition public (in the sense that they have to leave the wallet if you want to receive transactions from others), and there's no such a thing as a private address. so it would make sense to call 4... addresses primary address wherever they occur. (Zero To Monero sometimes calls them main addresses, but I haven't seen that anywhere else ever.)

and yes, we should definitely stop using primary addresses wherever possible, and just call subaddresses addresses. the distinction creates behavior that damages privacy (reusing the primary address because it seems more certain/official). you could remove it from the list of addresses people can receive to (Feather has already done this). however, I don't think you can get rid of it when exporting view-only credentials or creating a view-only wallet. that's because to generate subaddresses [you also need](https://web.getmonero.org/library/Zero-to-Monero-2-0-0.pdf#section.4.3) the public spend key (*K<sup>s</sup>*), and that's by convention transferred in the form of the primary address (*K<sup>v</sup>, K<sup>s</sup>*). you could start accepting/displaying only the public spend key instead, but people who want to import from or export to other wallets would be in trouble.
